homeassistant/util/enum.py

```diff
@@ -8,7 +8,7 @@
 class StrEnum(str, Enum):
     """Partial backport of Python 3.11's StrEnum for our basic use cases."""
 
-    def __new__(cls, value: str, *args: Any, **kwargs: Any) -> StrEnum:
+    def __new__(cls, value: str, *args: Any, **kwargs: Any):  # type: ignore[no-untyped-def]
```

Comment:
This could probably be resolved by using a `TypeVar`.

```py
from typing import TypeVar

T = TypeVar("T", bound="StrEnum")

class StrEnum(str, Enum):
    def __new__(cls: type[T], value: str, *args: Any, **kwargs: Any) -> T:
        ...
```

Reply:
I tried something like that, but Pylance didn't like that either

Reply:
Which file(s) did you have issues with? I just checked `esphome/button.py` and that seems to be fine. I can open a PR for it if you like.

Reply:
Pylance/Pyright complains directly in enum.py with the suggestion you've provided. As said, I tried the above as well.

Reply:
<img width="966" alt="Screen Shot 2021-11-30 at 14 47 15" src="https://user-images.githubusercontent.com/30130371/144058808-023fc223-36a9-42a7-9c97-3871b5d149f9.png">

At which point? This is with pyright set to `strict`. The error on `isinstance` is unrelated.

Reply:
![image](https://user-images.githubusercontent.com/195327/144060062-b4b7552d-5530-46ff-a3d6-b1a4a0b2d999.png)

Reply:
🤦